staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -266,7 +272,11 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope *RequestSc
 		admit = admission.WithAudit(admit, ae)
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := request.UserFrom(ctx)
-			attrs := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, "", scope.Resource, scope.Subresource, admission.Delete, dryrun.IsDryRun(options.DryRun), userInfo)
+			// For backwards compatibility, we don't change the options TypeMeta that are passed to rest storage Delete.
+			// But we do provide options to the admission webhooks with the correct TypeMeta.
```

Comment:
is there a reason we can't pin the DeleteOptions apiVersion/kind to meta.k8s.io/v1 once we successfully decode what the user provided?

Reply:
Nope, looks safe. Applying.